### PR TITLE
Added support to print failing testcase with its frequency

### PR DESCRIPTION
**Pull request by KeerthanaAP**
Fix https://github.com/ocp-power-automation/ci-monitoring-automation/issues/26


```
$python CI_JobHistory.py
--------------------------------------------------------------------------------------------------


1  4.11 libvirt
2  4.11 to 4.12 upgrade
3  4.12 libvirt
4  4.12 to 4.13 upgrade
5  4.12 heavy build
6  4.13 libvirt
7  4.13 powervs
8  4.13 to 4.14 upgrade
9  4.13 heavy build
10  4.14 libvirt
11  4.14 powervs
12  4.14 to 4.15 upgrade
13  4.14 heavy build
14  4.15 libvirt
15  4.15 powervs original
16  4.15 powervs siguid
17  4.15 heavy build
18  4.14 MCE
19  4.14 SNO
Select the required ci's serial number with a space 11
Enter Before date (YYYY-MM-DD): 2024-01-10
Enter After date (YYYY-MM-DD): 2024-01-01
Please select one of the option from Job History functionalities: 
1. Check Node Crash
2. Brief Job information
3. Detailed Job information
4. Failed testcases
5. Get builds with testcase failure
6. Get testcase frequency
Enter the option: 6
Checking runs from 2024-01-01 to 2024-01-10
Enter the testcase names comma seperated: [sig-instrumentation][Late] OpenShift alerting rules [apigroup:image.openshift.io] should have description and summary annotations [Skipped:Disconnected] [Suite:openshift/conformance/parallel],[sig-network][Feature:EgressFirewall] when using openshift ovn-kubernetes should ensure egressfirewall is created  [Suite:openshift/conformance/parallel]
['[sig-instrumentation][Late] OpenShift alerting rules [apigroup:image.openshift.io] should have description and summary annotations [Skipped:Disconnected] [Suite:openshift/conformance/parallel]', '[sig-network][Feature:EgressFirewall] when using openshift ovn-kubernetes should ensure egressfirewall is created  [Suite:openshift/conformance/parallel]']
-------------------------------------------------------------------------------------------------
4.14 powervs
+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-------------+
| Testcase                                                                                                                                                                                         |   Frequency |
+==================================================================================================================================================================================================+=============+
| [sig-instrumentation][Late] OpenShift alerting rules [apigroup:image.openshift.io] should have description and summary annotations [Skipped:Disconnected] [Suite:openshift/conformance/parallel] |           6 |
+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-------------+
| [sig-network][Feature:EgressFirewall] when using openshift ovn-kubernetes should ensure egressfirewall is created  [Suite:openshift/conformance/parallel]                                        |           1 |
+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-------------+




python CI_JobHistory.py
--------------------------------------------------------------------------------------------------


1  4.11 libvirt
2  4.11 to 4.12 upgrade
3  4.12 libvirt
4  4.12 to 4.13 upgrade
5  4.12 heavy build
6  4.13 libvirt
7  4.13 powervs
8  4.13 to 4.14 upgrade
9  4.13 heavy build
10  4.14 libvirt
11  4.14 powervs
12  4.14 to 4.15 upgrade
13  4.14 heavy build
14  4.15 libvirt
15  4.15 powervs original
16  4.15 powervs siguid
17  4.15 heavy build
18  4.14 MCE
19  4.14 SNO
Select the required ci's serial number with a space 11
Enter Before date (YYYY-MM-DD): 2024-01-23
Enter After date (YYYY-MM-DD): 2024-01-22
Please select one of the option from Job History functionalities: 
1. Check Node Crash
2. Brief Job information
3. Detailed Job information
4. Failed testcases
5. Get builds with testcase failure
6. Get testcase failure frequency
Enter the option: 6
Checking runs from 2024-01-22 to 2024-01-23
Enter the testcase names comma seperated (Leave it empty to fetch all tcs frequency): 
-------------------------------------------------------------------------------------------------
4.14 powervs
+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-------------+
| Testcase                                                                                                                                                                                                                           |   Frequency |
+====================================================================================================================================================================================================================================+=============+
| [sig-network][Feature:EgressFirewall] egressFirewall should have no impact outside its namespace [Suite:openshift/conformance/parallel]                                                                                            |           2 |
+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-------------+
| [sig-instrumentation][Late] OpenShift alerting rules [apigroup:image.openshift.io] should link to a valid URL if the runbook_url annotation is defined [Suite:openshift/conformance/parallel]                                      |           2 |
+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-------------+
| [sig-storage] In-tree Volumes [Driver: nfs] [Testpattern: Dynamic PV (default fs)] subPath should support file as subpath [LinuxOnly] [Suite:openshift/conformance/parallel] [Suite:k8s]                                           |           1 |
+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-------------+
| [sig-storage] PersistentVolumes NFS when invoking the Recycle reclaim policy should test that a PV becomes Available and is clean after the PVC is deleted. [Suite:openshift/conformance/parallel] [Suite:k8s]                     |           1 |
+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-------------+
| [sig-instrumentation][Late] OpenShift alerting rules [apigroup:image.openshift.io] should have a valid severity label [Skipped:Disconnected] [Suite:openshift/conformance/parallel]                                                |           1 |
+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-------------+
| [sig-storage] In-tree Volumes [Driver: nfs] [Testpattern: Dynamic PV (default fs)] volumes should store data [Suite:openshift/conformance/parallel] [Suite:k8s]                                                                    |           1 |
+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-------------+
| [sig-storage] CSI Mock volume fsgroup policies CSI FSGroupPolicy [LinuxOnly] should modify fsGroup if fsGroupPolicy=File [Suite:openshift/conformance/parallel] [Suite:k8s]                                                        |           1 |
+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-------------+
| [sig-storage] CSI Mock workload info CSI workload information using mock driver should be passed when podInfoOnMount=true [Suite:openshift/conformance/parallel] [Suite:k8s]                                                       |           1 |
+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-------------+
| [sig-instrumentation][Late] OpenShift alerting rules [apigroup:image.openshift.io] should link to an HTTP(S) location if the runbook_url annotation is defined [Suite:openshift/conformance/parallel]                              |           1 |
+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-------------+
| [sig-storage] CSI Mock volume storage capacity CSIStorageCapacity CSIStorageCapacity disabled [Suite:openshift/conformance/parallel] [Suite:k8s]                                                                                   |           1 |
+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-------------+
| [sig-storage] In-tree Volumes [Driver: nfs] [Testpattern: Generic Ephemeral-volume (default fs) (immediate-binding)] ephemeral should create read/write inline ephemeral volume [Suite:openshift/conformance/parallel] [Suite:k8s] |           1 |
+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-------------+
| [sig-storage] In-tree Volumes [Driver: nfs] [Testpattern: Dynamic PV (default fs)] subPath should support non-existent path [Suite:openshift/conformance/parallel] [Suite:k8s]                                                     |           1 |
+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-------------+
| [sig-storage] In-tree Volumes [Driver: nfs] [Testpattern: Generic Ephemeral-volume (default fs) (late-binding)] ephemeral should support multiple inline ephemeral volumes [Suite:openshift/conformance/parallel] [Suite:k8s]      |           1 |
+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-------------+
| [sig-arch] Managed cluster should set requests but not limits [Suite:openshift/conformance/parallel]                                                                                                                               |           1 |
+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-------------+
| [sig-instrumentation][Late] OpenShift alerting rules [apigroup:image.openshift.io] should have a runbook_url annotation if the alert is critical [Skipped:Disconnected] [Suite:openshift/conformance/parallel]                     |           1 |
+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-------------+
| [sig-instrumentation][Late] OpenShift alerting rules [apigroup:image.openshift.io] should have description and summary annotations [Skipped:Disconnected] [Suite:openshift/conformance/parallel]                                   |           1 |
+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-------------+

### CODE REVIEW

File: monitor.py [hunk 1]
```diff
@@ -428,6 +428,44 @@ def get_failed_monitor_testcases(spy_link,job_type):
...
+
+    """
+    frequency = {}
+    for spylink in spylinks:
+        job_type,_ = job_classifier(spylink)
+        lease,_ = get_quota_and_nightly(spylink)
+        if zone is not None and lease not in zone :
+            continue
+        cluster_status=cluster_deploy_status(spylink)
+        if cluster_status == 'SUCCESS' and "4.15" not in spylink:
```
Comment (SurajGudaji) [inline monitor.py:451]: why does this function not support 4.15 builds?

Comment (KeerthanaAP) [reply monitor.py]: Removed it here and get_failed_testcases() as well.

**Review by Shilpa-Gokul — APPROVED** [review]

**Pull request merged**